MapleServer2/Types/EffectTriggers.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MapleServer2.Types;
+
+public struct EffectTriggers
+{
+    public bool IsEvent;
+    public int SkillId;
+    public int EffectId;
+    public IFieldActor Owner;
+    public IFieldActor Target;
+    public IFieldActor Caster;
+
+    public EffectTriggers()
+    {
+        IsEvent = false;
+        SkillId = 0;
+        EffectId = 0;
+        Owner = null;
+        Target = null;
+        Caster = null;
```

Comment:
It's used for handling begin condition stuff. Owner, Target, and Caster are all different targets that can be checked for conditions, and IsEvent, SkillId, and EffectId are specifically for identifying conditions that should only happen on a certain skill use or effect. Certain conditions should fail if IsEvent is true to prevent double firing, while others should fail on false because they're meant to respond to events

Reply:
I understand that each variable has a purpose but you dont need to specify the values of these in the constructor because the default of SkillId is 0, default of Owner is null, etc etc.

Reply:
Ohh I see now. Yeah I didn't know you were referring to the constructor specifically.